### ISSUE TITLE
Boundary element method

### ISSUE THREAD

**Issue by jcharris** [mfem/mfem#88]
Has MFEM ever been used for a boundary element method code?  Would it be difficult for a person to make this modification, or is this not worth the effort?  It appears that it would be a question of rewriting a Bilinear form for the assembly and solve phases, at a minimum?

Have been developing a boundary element code for some years and working to update the higher-order elements, and it would be advantageous to have some of the features that are already developed in MFEM, I wasn't sure if it was easier to write from scratch the needed parts, or if it would be possible to make use of a code like this.

**Comment by mlstowell** [comment]
As far as I know MFEM has not been used for a boundary element code.

I have seen finite element libraries used for boundary element codes in the past and it is possible. The mesh and basis function classes could probably be used with very minor modifications.  A boundary element code would require much more advanced integration rules which could properly handle singular Green's function kernels. Typically, efficiently computing and possibly caching the jacobians is important in a boundary element code so this may need to be added.  Other issues I've seen deal with non-trivial topologies of surface meshes such as three elements meeting at a single edge in a T-shape.  Also, boundary element codes are typically written with complex valued fields whereas MFEM is real valued.

In short, it would require a great deal of specialized code but it should be possible for a subset of boundary element applications at least.
